client/client.go

```diff
@@ -47,12 +47,13 @@ type sendRequest struct {
 
 // Client implements an HTTP sender for the SAPM protocol
 type Client struct {
-	numWorkers  uint
-	maxIdleCons uint
-	endpoint    string
-	accessToken string
-	httpClient  *http.Client
-	closeCh     chan struct{}
+	numWorkers         uint
+	maxIdleCons        uint
+	endpoint           string
+	accessToken        string
+	httpClient         *http.Client
+	disableCompression bool
```

Comment:
Should we avoid negating? Would it make more sense to use `useCompression` boolean with default value of true?
Are we going to expose this option to end user? I think we should. In that case what will be the name of the config option and what will be the default value? I think we should keep the default value to match current behavior, so maybe what you did is better.

Reply:
yes, I was also thinking about it. disableCompression looks like a better option for end user, then I decided to be consistent and brought it here

Reply:
so for end use it would be `disable_compression` with `false` by default and usually not being used. What do you think?

Reply:
ok

Reply:
This is fine but I think `compression: "gzip"` as default would've been nicer. People could set it to empty string to disable compression altogether.